fuel-tx/src/transaction/fee.rs

```diff
@@ -113,18 +120,23 @@ impl TransactionFee {
     /// Attempt to create a transaction fee from parameters and transaction internals
     ///
     /// Will return `None` if arithmetic overflow occurs.
-    pub fn checked_from_tx<T: Chargeable>(
+    pub fn checked_from_tx<T: Chargeable + Inputs>(
+        gas_costs: &GasCosts,
```

Comment:
Regarding predicates, you can already add a new cost into `GasCosts` and add a new structure that we can use later to charge for the predicates. 

Maybe you want to extend `FeeParameters`, for example, the price of the `ecr1` can be duplicated there. The constructor can accept `gas_costs`, like `FeeParameters::new(gas_costs)`.

Reply:
We could add new fields for both the cost of signed input address recovery and the cost of predicate ownership check. The advantage of this is that we don't have to pass `GasCosts` around, which is a little bit simpler. 

On the other hand, it is not clear to me why that should be the responsibility of `FeeParameters` when `GasCosts` already contains these values. `FeeParameters` is responsible for providing information about how to convert gas to fees, rather than storing any concrete gas costs themselves.